web/src/utils/helpers.test.ts

```diff
@@ -1,59 +1,187 @@
-import { getFossilFuelPercentage } from './helpers';
+import {
+  getCO2IntensityByMode,
+  dateToDatetimeString,
+  getProductionCo2Intensity,
+  getFossilFuelRatio,
+  getCarbonIntensity,
+  getRenewableRatio,
+} from './helpers';
 
-describe('getFossilFuelPercentage', () => {
+import { zoneDetailMock } from 'stories/mockData';
+
+describe('getCO2IntensityByMode', () => {
+  // Tests for consumption
+  describe('consumption', () => {
+    it('returns 100 when the mode is consumption', () => {
+      const actual = getCO2IntensityByMode(
+        { co2intensity: 100, co2intensityProduction: 200 },
+        'consumption'
+      );
+      expect(actual).toBe(100);
+    });
+  });
+
+  // Tests for production
+  describe('production', () => {
+    it('returns 200 when the mode is production', () => {
+      const actual = getCO2IntensityByMode(
+        { co2intensity: 100, co2intensityProduction: 200 },
+        'production'
+      );
+      expect(actual).toBe(200);
+    });
+  });
+});
+
+describe('dateToDatetimeString', () => {
+  it('returns the correct datetime string', () => {
+    const actual = dateToDatetimeString(new Date('2023-01-01T12:00:00Z'));
+    expect(actual).toBe('2023-01-01T12:00:00Z');
+  });
+});
+
+describe('getProductionCo2Intensity', () => {
+  it('returns 10.7 when the type is hydro', () => {
+    const actual = getProductionCo2Intensity('hydro', zoneDetailMock);
+    expect(actual).toBe(10.7);
+  });
+
+  it('returns 136.442_667_362_438_53 when the type is battery storage', () => {
+    const actual = getProductionCo2Intensity('battery storage', zoneDetailMock);
+    expect(actual).toBe(136.442_667_362_438_53);
+  });
+});
```

Comment:
I don't think you need to add the correct value to the "title" here, as it just means having to change in two places if the number changes - and when running tests and getting an overview, we shouldn't have to care about what the specific value is :)

Perhaps we could name it something like `it('returns correct value for battery storage')` instead?

Reply:
I kind of forgot about this PR but I'll update the tiles and merge this. 😅